test/unit/BFactory.t.sol

```diff
@@ -207,24 +208,44 @@ contract BFactory_Unit_Collect is BFactoryTest {
   }
 
   /**
-   * @notice Test that the function fail if the transfer failed
+   * @notice Do not couple the collect logic to the known BToken
+   * implementation. Some ERC20's do not return a `bool` with to indicate
+   * success and only rely on reverting if there's an error. This test ensures
+   * we support them.
+   */
+  function test_Succeed_TransferNotReturningBoolean(address _lpToken, uint256 _toCollect) public {
+    assumeNotForgeAddress(_lpToken);
+
+    vm.mockCall(_lpToken, abi.encodeWithSelector(IERC20.balanceOf.selector, address(bFactory)), abi.encode(_toCollect));
+    vm.mockCall(_lpToken, abi.encodeWithSelector(IERC20.transfer.selector, owner, _toCollect), abi.encode());
+
+    vm.expectCall(_lpToken, abi.encodeWithSelector(IERC20.transfer.selector, owner, _toCollect));
+    vm.prank(owner);
+    bFactory.collect(IBPool(_lpToken));
+  }
+
+  /**
+   * @notice Test the function fails if the transfer failed
    */
   function test_Revert_TransferFailed(address _lpToken, uint256 _toCollect) public {
     assumeNotForgeAddress(_lpToken);
 
     vm.mockCall(_lpToken, abi.encodeWithSelector(IERC20.balanceOf.selector, address(bFactory)), abi.encode(_toCollect));
     vm.mockCall(_lpToken, abi.encodeWithSelector(IERC20.transfer.selector, owner, _toCollect), abi.encode(false));
 
-    vm.expectRevert(IBFactory.BFactory_ERC20TransferFailed.selector);
+    vm.expectRevert(abi.encodeWithSelector(SafeERC20.SafeERC20FailedOperation.selector, _lpToken));
     vm.prank(owner);
     bFactory.collect(IBPool(_lpToken));
   }
 }
```

Comment:
If we are going to use SafeERC20, I would just remove this test.

Reply:
not to bad to keep them imo